models/enterprise_sonic/mfa/sonic_mfa.yml

```diff
@@ -0,0 +1,139 @@
+              mfa_security_profile:
+                description:
+                  - Security profile contains the certificate for MFA service
+                type: str
+              client_secret:
+                description:
+                  - Encrypted password used in basic authorization header for MFA REST API
+                type: str
+              client_secret_encrypted:
+                description:
+                  - Indicates whether the client-secret is plain text or encrypted
+                type: bool
+          rsa_global:
+            description:
+              - RSA Global configuration
+            type: dict
+            suboptions:
+              rsa_security_profile:
+                description:
+                  - Security profile with CA-cert for validating RSA SecurID server
+                type: str
+          rsa_servers:
+            description:
+              - RSA Server configuration
+            type: list
+            elements: dict
+            suboptions:
+              hostname:
+                description:
+                  - RSA server's hostname or IP address
+                type: str 
+                required: True
+              server_port:
+                description:
+                  - Port number of the RSA SecurID server
+                  - Range 1025-49151
+                type: int
+              client_id:
+                description:
+                  - Unique identifier of the system as a client of SecurID service, assigned by SecurID service
+                type: str
+              client_key:
+                description:
+                  - Encrypted Key associated with the client-id, assigned by SecurID service
+                type: str
+              client_key_encrypted:
+                description:
+                  - Indicates whether the client-key is plain text or encrypted
+                type: bool
+              connection_timeout:
+                description:
+                  - Timeout in seconds for connection to the SecurID server
+                  - Range 1-30
+                type: int
+              read_timeout:
+                description:
+                  - Timeout in seconds to read from the SecurID server
+                  - Range 1-150
+                type: int
+          cac_piv_global:
+            description:
+              - CAC-PIV Global configuration
+            type: dict
+            suboptions:
+              cacpiv_security_profile:
+                description:
+                  - Security profile for SSH access with CAC-PIV
+                type: str
+              cert_username_field:
+                description:
+                  - SSH user certificate field for matching with SSH login username
+                type: str
+              cert_username_match:
+                description:
+                  - Match option to parse the username from respective certificate field
+                type: str
```

Comment:
Since only a set of pre-defined string values are supported for these options, those values can be listed as 'choices'.